lib/tom_queue/deferred_work_manager.rb

```diff
@@ -80,13 +80,19 @@ def schedule(response, headers, payload)
     def start
       debug "[DeferredWorkManager] Deferred process starting up"
 
+      register_signal_handlers
```

Comment:
Oh yeah, I know. The problem is that `main_loop` runs the main thread. So far the only way I can imagine is to handle it here.

Arguably do you think it's OK to run it in a new thread?